### ISSUE TITLE
Oboe library

### ISSUE THREAD

**Issue by DebugBSD** [SFML/SFML#1619]
Is there any chance to use Oboe library instead of OpenAL for Android platforms? It seems that OpenAL is outdated and I cannot find the source code to build it. I'm using Visual Studio 2019 to build SFML and I have done with window, graphics and system module but I'm stuck with audio module because of OpenAL. I would like to know if you are planning to use Oboe in a near future.

**Comment by LaurentGomila** [comment]
I didn't even know about that library. But since it's an Android specific lib, we're not going to use it unless this is the only choice for that platform.

For OpenAL, we use OpenAL Soft: https://github.com/kcat/openal-soft

Seems like it compiles fine (with or without hacks) on Android.

**Comment by DebugBSD** [comment]
OK, thank you very much any way.

PS: I tried to build OpenAL-Soft with CMake on Windows and I couldn't. I'd got a lot of problems. That's why I thought about move on to Oboe, because is new and it seems that every project is going to use it.  OpenAL is not used anymore on Android Platforms. Instead of OpenAL Android gives support to OpenSL and AAudio systems (Both of them, supported by Oboe).

**Comment by MarioLiebisch** [comment]
The problem is not about making the switch, it's also about implementing and maintaining yet another interface, especially considering it's Android only. We could also drop OpenAL on Windows in favor of DirectSound, but that would also come at an implementation and maintenance cost.

**However,** you could at any time clone SFML's repository and actually implement support for Oboe. Maybe there are others that would prefer using that over OpenAL, too.

**Comment by DebugBSD** [comment]
Yes, I thought about it. I know that is difficult to give support to several platforms at the same time with the same library. 

Anyway, I did a branch of SFML and began to port Oboe and Opus & OpusFile. 

Thank you very much.

**Comment by LaurentGomila** [comment]
OpenAL Soft seems to have an OpenSL backend, and there are references to Android in its CMake scripts and source code that aimply that it's compatible with that OS. You should investigate more before trying to make such a big change to SFML.

**Comment by DebugBSD** [comment]
Yes, I know that. 

First of all, I'm so sorry if this is not the correct place to say this. Maybe I should open a new thread in the forum. 

The thing is I tried to build OpenAL from Visual Studio but, it was impossible to me. When I ran cmake and I choose Visual Studio - ARM64 on the GUI I simply got an error so I couldn't build the library with CMake. Every thing I did to try to build the library was impossible. That's why I ended with OpenAL and I tried to port sfml-audio to Oboe.
